vapi/tags/tag_association.go

```diff
@@ -42,8 +42,8 @@ func (c *Manager) AttachTag(ctx context.Context, tagID string, ref mo.Reference)
 	if err != nil {
 		return err
 	}
-	spec := internal.NewAssociation(id, ref)
-	url := internal.URL(c, internal.AssociationPath).WithAction("attach")
+	spec := internal.NewAssociation(ref)
+	url := internal.URL(c, internal.AssociationPath).WithID(id).WithAction("attach")
```

Comment:
Tested `govc` based on this commit against VC and it works. As this is an internal API, we're not breaking any users of the tagging client, e.g. vSphere Zones implementation in K8s.